File: src/disaggregated_component/memory_module/memory_sub_module.cc
#include "disaggregated_component/memory_module/memory_sub_module.hh"

MemorySubModule::MemorySubModule(Params *p):
    BaseModule(p)
{}

MemorySubModule*
MemorySubModuleParams::create()
{
    return new MemorySubModule(this);
}

void
MemorySubModule::init()
{
    // removed system port check
    return;
}

File: src/disaggregated_component/memory_module/MemorySubModule.py
from m5.params import *
from m5.proxy import *
from m5.objects.BaseModule import BaseModule


class MemorySubModule(BaseModule):
    type = 'MemorySubModule'
    cxx_header = "disaggregated_component/memory_module/memory_sub_module.hh"
    is_dummy = True
